LoopKit/WeakObserverSet.swift

```diff
@@ -8,41 +8,41 @@
 
 import Foundation
 
-public struct WeakObserverSetIterator<Observer>: IteratorProtocol {
-    let enumerator: NSEnumerator
-    
-    init(_ enumerator: NSEnumerator) {
-        self.enumerator = enumerator
-    }
-    
-    public mutating func next() -> Observer? {
-        return enumerator.nextObject() as? Observer
+public class WeakObserverSet<Observer>: Sequence {
+
+    public struct Iterator<Observer>: IteratorProtocol {
```

Comment:
With `Iterator` declared as an inner struct, you can drop the generic parameter and it will inherit it from the outer scope.